hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/DefaultSource.scala

```diff
@@ -56,6 +56,9 @@ class DefaultSource extends RelationProvider
       // Enable "passPartitionByAsOptions" to support "write.partitionBy(...)"
       spark.conf.set("spark.sql.legacy.sources.write.passPartitionByAsOptions", "true")
     }
+    // Revisit EMR Spark and EMRFS incompatibilities, for now disable
+    spark.conf.set("spark.sql.dataPrefetch.enabled", "false")
+    spark.sparkContext.hadoopConfiguration.set("fs.s3.metadata.cache.expiration.seconds", "0")
```

Comment:
Shouldn't this be guarded by a Hudi config as it is an issue on EmrFs?
Also, I did not find any documentation for `fs.s3.metadata.cache.expiration.seconds` config in https://hadoop.apache.org/docs/stable/hadoop-aws/tools/hadoop-aws/s3guard.html?
Have we tested this change?

Reply:
@codope Yes the property is specific to `EmrFS` and its not present in open source implementations. This is done as a temporary fix to make sure Hudi 0.12 can work out of the box on EMR environment. This property should act as No Op for other environments.

We are also working internally to have a fix for this on EmrFS side instead, and should get rid of this in next release. It was tested using EMR's internal integration tests. With the current hudi master it was breaking several of our tests, and with this fix we get past them.

Reply:
Instead of hard-coding to false, shall we also allow for override using --conf . I wonder if it's better todo this in the EMR Huds configs, rather than code?

Reply:
Well the only reason we did this is because we want to reduce the noise, for customers having to pass additional configurations just to make things work on EMR. We cannot store in EMR Hudi configs, because as of now the global Hudi confs that we support only work for Hudi related configurations. We cannot pass spark/hadoop configs in them.

If you guys have concerns about this, we can revert it and instead have it in the documentation that customers should explicitly pass these when running open source bundle on EMR. Its just that it is not a good experience.

Reply:
Let's prioritize user experience and keep it in this release for now. Tracking the removal of these configs after proper fix in HUDI-4473

Reply:
Let's prioritize user experience and keep it in this release for now. @umehrot2 already created a tracking issue to revert it in 0.13.0 HUDI-4470